app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,36 @@
+class VotesController < ApplicationController
+  def create
+    if session[:user_id]
+      user = User.find_by(id: session[:user_id])
+      work = Work.find_by(id: params[:work_id])
+      user_votes = user.votes
+      work_votes = work.votes
+
+      user_votes.each do |vote|
+        if work_votes.include?(vote)
+          flash[:status] = :error
+          flash[:message] = "You have already voted for this media"
```

Comment:
Instead of a loop here, you could say `if Vote.find_by(user_id: user.id, work_id: work.id)`

Reply:
Would it be possible to put this check in the model, possibly as a validation?